dashboard/client/src/service/task.ts

```diff
@@ -18,7 +18,7 @@ export const getTask = (taskId: string) => {
 };
 
 export const downloadTaskTimelineHref = (jobId: string | undefined) => {
-  let url = "/api/v0/tasks/timeline?download=1";
+  let url = "api/v0/tasks/timeline?download=1";
```

Comment:
This fix is fragile and will not work if we move away from "hash-based" routing to location based routing for the dashboard UI.

If we move to location based routing (standard for most single page apps), the path for the the job detail page will be `dashboard.com/jobs/{job_id}`, and thus, this link will go to `dashboard.com/jobs/{job_id}/api/v0/tasks/timeline?download=1` instead of `dashboard.com/api/v0/tasks/timeline?download=1`

I think the proper fix is to add an environment variable to dashboard_head.py called `RAY_DASHBOARD_PATH_PREFIX`. 
The path_prefix must then be injected into the web app and used throughout the code base.

Then user can launch ray like `RAY_DASHBOARD_PATH_PREFIX=ray/dashboard ray start --head`

Reply:
Looking at our other apis, we follow this same pattern... Seems like we'd have to do a major migration if we moved from hash-based routing to location-based routing. I'll approve this then since this is the existing pattern.

Reply:
Yeah, the hash-based routing is a major new feature and need more works to do. Maybe we should submit a REP first, and any suggestions? @alanwguo